### PR TITLE
Uses master as initial branch when cloning secure_config_repo

### PR DESCRIPTION
**Pull request by pomegranited**
If there happens to be no 'release' branch present in the `secure_config_repo`, the tasks.yml playbook fails when cloning the `secure_config_repo`.

**JIRA tickets**: Cleanup for [OLIVE-22](https://openedx.atlassian.net/browse/OLIVE-20), [OC-1512](https://tasks.opencraft.com/browse/OC-1512)

**Dependencies**: None

**Sandbox URL**: 

http://52.20.136.133:8080/ - ping me for login credentials.

**Testing instructions**:

See [edx/configuration PR #2939](https://github.com/edx/configuration/pull/2939) for full test instructions.

Expected result is the remote task will successfully clone the secure repo (if your configuration is correct).

Without this fix, if you happen to have no branch named 'release' in your secure config, the tasks.yml fails with:

```
TASK: [secure config repository checked out] ********************************** 
failed: [172.30.2.184] => {"failed": true}
msg: Failed to checkout release
```

**Reviewers**
- [x] @jbzdak 
- [x] @itsjeyd  
- [x] @mulby

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=232&repo=edx%2Fedx-analytics-pipeline

**Comment by pomegranited** [comment]
@jbzdak  @mulby This PR is ready for review.

**Comment by jbzdak** [comment]
@pomegranited Formally `master` is not a special name for git, there might be a repository without `master` branch (albeit it is much less probable than missing `release`), could you try omitting `version` key which would just download current HEAD (which must be defined).

**Comment by jbzdak** [comment]
:+1:

**Comment by mulby** [comment]
@itsjeyd looked into this rather extensively - he should review this as well

File: share/task.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@
       when: ansible_distribution in common_redhat_variants
       
     - name: analytics tasks repository checked out
-      git: repo={{ repo }} dest={{ working_repo_dir }} version=master
```
Comment (itsjeyd) [inline share/task.yml:109]: @pomegranited This introduces problems for acceptance tests. With default settings, this playbook checks out the tasks repo from `/edx/app/analytics_pipeline/analytics_pipeline`. If that repo is at a branch that contains a slash, this task fails. For example, for `gabe/fix-local-acceptance` I get the following when running acceptance tests:

```
TASK: [analytics tasks repository checked out] ******************************** 
failed: [localhost] => {"failed": true}
msg: Failed to checkout branch fix-local-acceptance
```

In principle, this problem can be solved by [injecting the name of the remote](https://github.com/open-craft/edx-analytics-pipeline/commit/f6a2ca5d38ec72cc2129a652db356a572c43c3d1) into the value of `version`, but this change was rejected when it was proposed as part of https://github.com/edx/edx-analytics-pipeline/pull/204. (Relevant discussion starts [here](https://github.com/edx/edx-analytics-pipeline/pull/204#discussion-diff-55208168).)

So I think we need to set `version` here, and `master` is probably our best bet in terms of a default value.

**Comment by pomegranited** [comment]
@itsjeyd Excellent, thank you for the recommendation.  

@mulby Fixed, and rebased.  Ok to merge?

**Comment by mulby** [comment]
:+1:

**Pull request merged**